packages/yourloops/pages/hcp/team-edit-modal.tsx

```diff
@@ -0,0 +1,240 @@
+import Modal from "@material-ui/core/Modal";
+import Select from "@material-ui/core/Select";
+import TextField from "@material-ui/core/TextField";
+
+import locales from '../../../../locales/languages.json';
+import { t } from "../../lib/language";
+import { REGEX_EMAIL } from "../../lib/utils";
+import { Team } from "../../models/team";
+import { useAuth } from "../../lib/auth/hook/use-auth";
+
+interface LocalesCountries {
+  [code: string]: {
+    name: string;
+  };
+}
+
+interface TeamEditModalProps {
+  action: "edit" | "create";
+  team: Partial<Team>;
+  modalOpened: boolean;
+  setModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  onSaveTeam: (team: Partial<Team>) => Promise<void>;
+}
+
+const modalStyles = makeStyles((theme: Theme) => {
+  return {
+    modal: {
+      display: "flex",
+      justifyContent: "center",
+      alignItems: "center",
+    },
+    divMain: {
+      display: "block",
+      padding: theme.spacing(2),
+      backgroundColor: theme.palette.background.paper,
+      borderRadius: theme.shape.borderRadius,
+      width: theme.breakpoints.width("sm"),
+    },
+    form: {
+      display: "flex",
+      flexDirection: "column",
+      padding: theme.spacing(2),
+      maxHeight: "30em",
+      overflowY: "scroll",
+    },
+    formChild: {
+      marginBottom: theme.spacing(2),
+    },
+    divModalButtons: {
+      display: "flex",
+      flexDirection: "row",
+      marginTop: theme.spacing(2),
+      marginRight: theme.spacing(4), // eslint-disable-line no-magic-numbers
+    },
+    divModalButtonCancel: {
+      marginLeft: "auto",
+      marginRight: theme.spacing(1),
+    },
+  };
+});
+
+function TeamEditModal(props: TeamEditModalProps): JSX.Element {
+  const { action, team, modalOpened, setModalOpen, onSaveTeam } = props;
+  const modalBackdropTimeout = 300;
+
+  const classes = modalStyles();
+  const auth = useAuth();
+
+  console.info("TeamEditModal", auth.user?.settings?.country);
+
+  const [ teamName, setTeamName ] = React.useState(team.name ?? "");
+  const [ teamPhone, setTeamPhone ] = React.useState(team.phone ?? "");
+  const [ teamEmail, setTeamEmail ] = React.useState(team.email ?? "");
+  const [ addrLine1, setAddrLine1 ] = React.useState(team.address?.line1 ?? "");
+  const [ addrLine2, setAddrLine2 ] = React.useState(team.address?.line2 ?? "");
+  const [ addrZipCode, setAddrZipCode ] = React.useState(team.address?.zip ?? "");
+  const [ addrCity, setAddrCity ] = React.useState(team.address?.city ?? "");
+  const [ addrCountry, setAddrCountry ] = React.useState(auth.user?.settings?.country ?? "FR");
+  const [ formIsIncomplete, setFormIsIncomplete ] = React.useState(true);
+
+  const ariaModal = action === "create" ? t("aria-modal-team-add") : t("aria-modal-team-edit");
+  const modalTitle = action === "create" ? t("modal-team-add-title") : t("modal-team-edit-title");
+  const modalButtonValidate = action === "create" ? t("modal-team-button-create") : t("modal-team-button-edit");
```

Comment:
for sure we are saving lines ;-) 
but I really prefer it written that way. 
```suggestion
  let ariaModal = t("aria-modal-team-add");
  let modalTitle = t("modal-team-add-title");
  let modalButtonValidate = t("modal-team-button-create");
  if (action !== "create") {
    ariaModal = t("aria-modal-team-edit");
    modalTitle = t("modal-team-edit-title");
    modalButtonValidate = t("modal-team-button-edit");
  }
```